test/lib/jdk/test/lib/thread/VThreadRunner.java

```diff
@@ -140,4 +142,41 @@ public static void run(int characteristics, ThrowingRunnable task) throws Except
     public static void run(ThrowingRunnable task) throws Exception {
         run(null, 0, task);
     }
+
+    /**
+     * Ensures that the virtual thread scheduler's target parallelism is at least the
+     * given size. If the current parallelism is less than size then it is changed to
+     * size. This method returns an AutoCloseable, its close method restores the
+     * parallelism.
+     *
+     * @return an object to restore the parallelism
+     */
+    public static AutoCloseable ensureParallelism(int size) {
+        ForkJoinPool pool;
+        try {
+            var clazz = Class.forName("java.lang.VirtualThread");
+            var field = clazz.getDeclaredField("DEFAULT_SCHEDULER");
+            field.setAccessible(true);
+            pool = (ForkJoinPool) field.get(null);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        int parallelism = pool.getParallelism();
+        if (parallelism >= size) {
+            return () -> { };
+        } else {
+            pool.setParallelism(size);
+            return new AutoCloseable() {
+                boolean closed;
+                @Override
+                public void close() {
+                    if (!closed) {
+                        closed = true;
+                        pool.setParallelism(parallelism);
```

Comment:
What is the rationale for restoring the parallelism? It's just a test. Is this really necessary? Are we reusing the JVM to run other tests, and even if we are does it matter?

Reply:
Look at the ThreadAPI test as an example. It's a TestNG with dozens of test methods. All but one can run on single core systems.

Reply:
Ok.